src/overlay/Peer.cpp

```diff
@@ -1820,6 +1826,11 @@ Peer::recvHello(Hello const& elo)
         return;
     }
 
+    if (mRemoteOverlayVersion >= Peer::FIRST_VERSION_REQUIRING_FLOW_CONTROL)
```

Comment:
Not following the reasoning for this if condition: remote version must be at least min version. min version right now is 23, so the condition should always be true.

Reply:
I removed some logic around the version as the minimum version must enable flow control by default. I removed this code during the cleanup.